src/eventlistener.cpp

```diff
@@ -49,21 +51,44 @@ EventListener::EventListener() {
     QLocalSocket* socket = m_server.nextPendingConnection();
     Q_ASSERT(socket);
 
-    connect(socket, &QLocalSocket::readyRead, socket, [socket]() {
-      QByteArray input = socket->readAll();
-      input = input.trimmed();
+    connect(socket, &QLocalSocket::readyRead, this,
+            &EventListener::socketReadyRead);
+    connect(socket, &QLocalSocket::disconnected, socket, &QObject::deleteLater);
```

Comment:
My understanding is that the socket is being deleted to free memory, although it would have been deleted when the parent QLocalServer was destroyed. Is that correct? Are we sure that the socket doesn't need to be used after it is disconnected (for example, to check connected state)? I suppose deleteLater will prevent most such cases.

Will the following be afffected by the socket being prematurely deleted on disconnect?

  `socket.disconnectFromServer();
  if (socket.state() != QLocalSocket::UnconnectedState) {
    socket.waitForDisconnected(1000);
  }``

Reply:
The leak seems to have been a bug in the code that existed before I got here, in particular the `QLocalServer` lives for the duration of the client UI instance, but the `QLocalSocket` connections are made for each message sent into the pipe. Failing to destroy them upon disconnection leaves them in a disconnected state that hang around effectively indefinitely and waste resources.

The `socket.disconnectFromServer()` function you mentioned is a separate socket in a separate process (the sender of the command). It will not be affected by the garbage collection of a disconnected socket in the receiver process.